### PR TITLE
Add MoviePlaybackSample for Swift 2.2

### PR DESCRIPTION
**Pull request by Wei-Xia**

### CODE REVIEW

**Comment by toshi0383** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by toshi0383** [comment]
Tested on Xcode7.3.1. TVMLKit fails to parse the xmlString. Could you look at it?
```
2017-01-07 08:19:15.763 MoviePlaybackSample[6519:228833] ITML <Error>: Error Domain=ITMLKitErrorDomain Code=200 "Entity: line 1: parser error : Start tag expected, '<' not found
Optional(<3c3f786d 6c207665 7273696f 6e3d2231 2e302220 656e636f 64696e67 3d22555
^
```

File: MoviePlaybackSample/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+    }
+    
+    func applicationDidBecomeActive(application: UIApplication) {
+        // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+    }
+    
+    func applicationWillTerminate(application: UIApplication) {
+        // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+    }
```
Comment (toshi0383) [inline MoviePlaybackSample/AppDelegate.swift]: Functions above are not necessary. Could you delete these lines?

File: MoviePlaybackSample/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    let data = NSData(contentsOfURL: NSURL(fileURLWithPath: NSBundle.mainBundle().pathForResource("Sample", ofType: "xml")!))
+    
+    return String(data)
+}
+
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+    
+    var window: UIWindow?
```
Comment (toshi0383) [inline MoviePlaybackSample/AppDelegate.swift]: This property is not necessary.

File: MoviePlaybackSample/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        default:
+            return nil
+        }
+    }
+}
+
+func startPlayback(url: NSURL) {
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0))
+    {
```
Comment (toshi0383) [inline MoviePlaybackSample/AppDelegate.swift]: Could you put opening brace in same line?

File: MoviePlaybackSample/AppDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        let vc = AVPlayerViewController()
+        
+        vc.player = AVPlayer(URL: url)
+        
+        dispatch_async(dispatch_get_main_queue())
+        {
```
Comment (toshi0383) [inline MoviePlaybackSample/AppDelegate.swift]: Could you put opening brace in same line?

**Review by toshi0383 — CHANGES_REQUESTED** [review]

**Comment by Wei-Xia** [comment]
Hi @toshi0383, I have changed my code as your suggestions.

About error, YES, I do receive error message from Xcode 7.3 as below.
```
2017-01-06 21:32:18.596 MoviePlaybackSample[1380:18852] ITML <Error>: Error Domain=ITMLKitErrorDomain Code=200 "Entity: line 1: parser error : Start tag expected, '<' not found
Optional(<3c3f786d 6c207665 7273696f 6e3d2231 2e302220 656e636f 64696e67 3d22555
^
" UserInfo={NSLocalizedDescription=Entity: line 1: parser error : Start tag expected, '<' not found
Optional(<3c3f786d 6c207665 7273696f 6e3d2231 2e302220 656e636f 64696e67 3d22555
^
} - file:///Users/weixia/Library/Developer/Xcode/DerivedData/TVMLKitchen-dfsluypjpahpfkglmlxkmjhoonyi/Build/Products/Debug-appletvsimulator/TVMLKitchen.framework/kitchen.js - line:119:37
```

However, I use the exactly same code in my own project with `swift 2.2`, which works perfectly correctly. 
But in this `MoviePlaybackSample`, it can't work. That's really confusing. I will try my best to figure out how to solve this issue.

**Comment by toshi0383** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by toshi0383** [comment]
@Wei-Xia 
Thanks for update!
As you can see from the error message,
```
Optional(<3c3f786d 6c207665 7273696f 6e3d2231 2e302220 656e636f 64696e67 3d22555
^
```
The bug is caused by this line. 
```
    return String(data)
```
This produces a String like `Optional(<0xdatadatadata...`, not correct String representation.
Please fix it to parse NSData correctly.
```
    return NSString(data: data!, encoding: NSUTF8StringEncoding)! as String
```

**Comment by toshi0383** [comment]
I closed and reopened the PR by mistake. No offense...

**Comment by Wei-Xia** [comment]
@toshi0383 

Yes, the error message is caused by `string`. I have changed as your suggestion, and it runs very well.

I don't know why the checks by Github/Bitrise was failed, but I can run it successfully locally.

**Comment by Wei-Xia** [comment]
I downloaded the repo from Github directly, and tried to run it.

It showed the error message below.
screen shot 2017-01-09 at 23 11 43

I believe this is the main reason causing failure in Github/Bitrise.

But from my own repository which I commit to this PR, there is no such error message happened. 

I will check with it tomorrow. 🤔

**Comment by toshi0383** [comment]
@Wei-Xia 
Thank you very much for your work. It looks great!
I currently set the default VM to be Xcode8.1, so it's ok for Bitrise failure for swift2.2 branch.

p.s. Bitrise does not handle different version of Xcodes at a time. You have to set manually from the web console, then trigger rebuild. And unfortunately, only the admin can see the web console, which is only me for this repo..  So you don't have to worry about the CI 😉

**Comment by toshi0383** [comment]
> I downloaded the repo from Github directly, and tried to run it.

Try building `TVMLKitchen` first before building `MoviePlaybackSample`.

**Pull request merged**